doc_src/language.rst

```diff
@@ -107,8 +107,7 @@ Some characters cannot be written directly on the command line. For these charac
 - ``\r`` represents the carriage return character.
 - ``\t`` represents the tab character.
 - ``\v`` represents the vertical tab character.
-- ``\xHH``, where ``HH`` is a hexadecimal number, represents the ASCII character with the specified value. For example, ``\x9`` is the tab character.
-- ``\XHH``, where ``HH`` is a hexadecimal number, represents a byte of data with the specified value. If you are using a multibyte encoding, this can be used to enter invalid strings. Only use this if you know what you are doing.
+- ``\xHH`` or ``\XHH``, where ``HH`` is a hexadecimal number, represents a byte of data with the specified value. For example, ``\x9`` is the tab character. If you are using a multibyte encoding, this can be used to enter invalid strings. Typically fish is run with the ASCII or UTF-8 encoding, so anything up to ``\X7f`` is an ASCII character.
```

Comment:
While we're cleaning the docs up here, can we just change "If you are using a multibyte encoding, this can be used to enter invalid strings." to "Note that this can be used to create invalid strings." ?

The term "multibyte encoding" [is rather ambiguous](https://stackoverflow.com/questions/748965/what-is-a-multibyte-character-set) as it has been used to refer to "legacy" non-Unicode encodings, modern variable-byte encodings like UTF-8 and UTF-16, or encodings that always take more than one byte like UTF-32.

It also depends on what "invalid string" means, too. Is it one that isn't portable, one that doesn't display on your terminal, one that isn't mapped to a character in your current codepage, or one that includes control characters. 

(Also, note that there's no good reason why we only warn about invalid strings for `\x` and `\X` but not for `\o`, `\u`, and `\U` but I don't know what we should do about that.)

Reply:
>(Also, note that there's no good reason why we only warn about invalid strings for \x and \X but not for \o, \u, and \U but I don't know what we should do about that.)

All three of those are interpreted as codepoint numbers.

That means the error case is much smaller - if `\u1234` is valid, then `\u1235` will also be, except in the single case of `\U10FFFF`.

In comparison, for `\x`/`\X` it depends much more immediately on the system's encoding, and for UTF-8 especially there are a lot of invalid code unit values, even between valid ones.

So it is much easier to end up with nonsense, and that seems worth a note. We can add another note that `\U10FFFF` is the highest valid unicode value, sure.

Reply:
Note that the case for Unicode values isn't as clear-cut. There are surrogate values that are valid Unicode codepoints but reserved for UTF-16 (0xD800 to 0xDFFF), so using them in a UTF-8 locale would result in an invalid string. I think there are other codepoints (not just byte values) that may be invalid depending on the particular encoding in use.

You're right though, I believe we already cap ingested octal escapes to saturate at 128, otherwise we'd have a problem with any single-byte octal-escaped value over 0o200 (all the way up to 0o777).

Reply:
I added in the maximum values.

Tbh I'd be entirely okay if we unified \u and \U as well.

I don't see the point in octal escapes *at all*, but removing those would be an actual backwards-compatibility concern. Put it on the "if I made fish from scratch" pile, I guess, along with `\v` (has anyone ever used that? who even knows what a "vertical tab" *is*?)

Reply:
also `\b`, I never used that intentionally but I often want to pass the literal backslash+b to `rg`